internal/metamorph/processor.go

```diff
@@ -615,3 +605,20 @@ func (p *Processor) Health() error {
 
 	return nil
 }
+
+func (p *Processor) storeData(ctx context.Context, data *store.StoreData) error {
```

Comment:
This function seems to be never returning an error

How about this as an alternative?

```
func (p *Processor) storeData(ctx context.Context, data *store.StoreData) error {
	nowNum := p.now()
	data.LastSubmittedAt = nowNum

	err := p.store.Set(ctx, data.Hash[:], data)
	if err != nil {
		return err
	}

	return nil
}
```

and when it is used on L 524

```
err = p.storeData(ctx, req.Data)
if err != nil {
     p.logger.Error("Failed to store transaction", slog.String("hash", data.Hash.String()), slog.String("err", err.Error()))
}

```

And on L 550 the function has to return with an error, because that is when the tx is stored initially. 
I see that this is a different logic than before, but I think it should be corrected like that.

```
err = p.storeData(ctx, req.Data)
if err != nil {
     return fmt.Errorf("failed to store transaction: %v", err)
}
```

Reply:
I believe it returns the error; the error is simply logged inside the function if it appears. If you are sure I should change the logic in line 550, I'll change it according to your suggestion

Reply:
Yes, when on L 550 storing the data fails, then we should return there with an error. So that the user knows something is wrong with ARC in case of the DB not being available. Otherwise the user thinks it's all fine and keeps submitting transactions.

Reply:
Changed